x/txfees/keeper/txfee_filters/arb_tx.go

```diff
@@ -195,3 +257,29 @@ func isAffiliateSwapMsg(msg []byte) bool {
 
 	return true
 }
+
+// check if this: https://celatone.osmosis.zone/osmosis-1/txs/8D20755D4E009CB72C763963A76886BCCCC5C2EBFC3F57266332710216A0D10D
+func isContractSwapContractMsg(msg []byte) bool {
+	// Check that the contract message is a valid JSON object
+	jsonObject := make(map[string]interface{})
+	err := json.Unmarshal(msg, &jsonObject)
+	if err != nil {
+		return false
+	}
+
+	// check the main key is "swap"
+	swap, ok := jsonObject["swap"].(map[string]interface{})
+	if !ok {
+		return false
+	}
+
+	if input_coin, ok := swap["input_coin"].(map[string]interface{}); !ok || len(input_coin) == 0 {
+		return false
+	}
+
+	if outputDenom, ok := swap["output_denom"].(string); !ok || len(outputDenom) == 0 {
+		return false
+	}
+
+	return true
```

Comment:
If we want to be more strict here we can check that the input and output denoms are either (1) the same, or (2) in a list of equivalent denoms (usdc/usdt/axlusdc/etc)

Reply:
This should work for now, but we are considering any contract swap as an attack, instead of just arbs